OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -132,10 +116,22 @@ public bool IsValidImpact(WPos pos, Actor firedBy)
 				return false;
 
 			var impactType = GetImpactType(world, targetTile, pos, firedBy);
-			if (!ValidImpactTypes.HasFlag(impactType) || InvalidImpactTypes.HasFlag(impactType))
-				return false;
+			var validImpact = false;
```

Comment:
This can be simplify down to:
```csharp
var validImpact = false;
switch (impactType)
{
	case ImpactType.TargetHit:
		return true;
	case ImpactType.Air:
		return IsValidTarget(new string[] { "Air" });
	case ImpactType.Ground:
		var tileInfo = world.Map.GetTerrainInfo(targetTile);
		return IsValidTarget(tileInfo.TargetTypes);
	default:
		return false;
}
```